### PR TITLE
feat: health check

### PR DESCRIPTION
**Pull request by aloknerurkar**
## Describe your changes

Add proper health check for mev-commit and oracle.

## Issue ticket number and link
#277 

## Checklist before requesting a review

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have made corresponding changes to the documentation
- [ ] I have tested this code by deploying the infrastructure and ensuring that commitments are being settled

### CODE REVIEW

File: oracle/pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -261,6 +263,20 @@ func (s *Service) RegisterMetricsCollectors(cs ...prometheus.Collector) {
...
+func (s *Service) RegisterHealthCheck(hc health.Health) {
+	s.router.HandleFunc(
+		"/health",
+		func(w http.ResponseWriter, r *http.Request) {
+			if err := hc.Health(); err != nil {
+				http.Error(w, err.Error(), http.StatusServiceUnavailable)
+				return
+			}
+			w.WriteHeader(http.StatusOK)
+			fmt.Fprintln(w, "ok")
```
Comment (Mikelle) [inline oracle/pkg/apiserver/apiserver.go:275]: Is it leftover after debug?

**Review by Mikelle — APPROVED** [review]

**Review by mrekucci — COMMENTED** [review]
The health check should also be used across infra Nomad scripts.

File: infrastructure/nomad/playbooks/templates/jobs/mev-commit-oracle.nomad.j2 [hunk 2]
```diff
@@ -42,6 +42,15 @@ job "{{ job.name }}" {
       port = "{{ port_name }}"
       tags = ["{{ port_name }}"]
       provider = "nomad"
+      {% if port_name == "http" %}
+      check {
+        name     = "HTTP Health Check"
```
Comment (mrekucci) [inline infrastructure/nomad/playbooks/templates/jobs/mev-commit-oracle.nomad.j2:47]: The name doesn't make much sense when there is no service name in it, so I would suggest omitting it. The same is below.

File: p2p/pkg/node/node.go [hunk 3]
```diff
@@ -109,6 +110,7 @@ func NewNode(opts *Options) (*Node, error) {
 
 	srv := apiserver.New(opts.Version, opts.Logger.With("component", "apiserver"))
 	peerType := p2p.FromString(opts.PeerType)
+	healthChecker := health.New()
```
Comment (mrekucci) [inline p2p/pkg/node/node.go:113]: nit: this can be defined closer to it's first usage on the line 547.

File: x/health/health.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
+package health
+
+import (
+	"fmt"
+
+	"google.golang.org/grpc/connectivity"
+)
+
+type HealthCheck interface {
```
Comment (mrekucci) [inline x/health/health.go:9]: `HealthCheck` -> `HealthChecker`

File: x/health/health.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+func GrpcGatewayHealthCheck(conn GrpcClientConn) HealthCheck {
+	return HealthCheckFunc(func() error {
+		if conn.GetState() == connectivity.TransientFailure {
+			return fmt.Errorf("grpc gateway: %s", conn.GetState())
+		}
+		return nil
+	})
+}
+
+func New() Health {
```
Comment (mrekucci) [inline x/health/health.go:54]: This constructor is not necessary, I'd suggest to export the `health` struct below and use `new(Health)` instead.

**Pull request merged**